modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v2_0/Swagger2Translator.java

```diff
@@ -581,4 +617,523 @@ public static Swagger getSwagger(Definition definition) {
+            property.setMinOccurs(swaggerProperty.getRequired() ? 1 : 0);
+            property.setMaxOccurs(swaggerProperty instanceof ArrayProperty ? -1 : 1);
+            property.setName(key);
+            // TODO: sub-properties not implemented in Swagger 2.0
+            // TODO: uniqueItems not implemented in Swagger 2.0
+            property.setUniqueItems(false);
+
+            if (swaggerProperty instanceof AbstractNumericProperty) {
+                AbstractNumericProperty numericProperty = (AbstractNumericProperty) swaggerProperty;
+                formatPropertyMinMax(numericProperty, property);
+            } else if (swaggerProperty instanceof ArrayProperty) {
+                ArrayProperty arrayProperty = (ArrayProperty) swaggerProperty;
+                property.setExample(arrayProperty.getItems().getExample());
+            }
+
+            property.setType(SwaggerTypes.toDefinitionType(swaggerProperty));
+
+            representation.getProperties().add(property);
+        }
+    }
+
+    /**
+     * Sorts the properties by their positions.
+     * 
+     * @param model
+     *            The model in which to sort the properties.
+     */
+    private static void sortSwaggerPropertiesByPosition(Model model) {
+        List<Map.Entry<String, Property>> propertyList =
+                new LinkedList<Map.Entry<String, Property>>(model.getProperties().entrySet());
+
+        Collections.sort(propertyList, new Comparator<Map.Entry<String, Property>>() {
+            public int compare(Map.Entry<String, Property> o1,
+                    Map.Entry<String, Property> o2) {
+                if (o1.getValue().getPosition() == null) {
+                    return -1;
+                }
+                return (o1.getValue().getPosition()).compareTo(o2.getValue().getPosition());
+            }
+        });
+
+        model.setProperties(new LinkedHashMap<String, Property>());
+        for (Iterator<Map.Entry<String, Property>> it = propertyList.iterator(); it.hasNext();) {
+            Map.Entry<String, Property> entry = it.next();
+            model.getProperties().put(entry.getKey(), entry.getValue());
+        }
+    }
+
+    /**
+     * Removes the dots in maximums and minimums for integer and long properties.
+     * 
+     * @param swaggerProperty
+     *            The Swagger property to process.
+     * @param property
+     *            The RWADef property to update
+     */
+    private static void formatPropertyMinMax(AbstractNumericProperty swaggerProperty,
+            org.restlet.ext.apispark.internal.model.Property property) {
+        String min = SwaggerUtils.toString(swaggerProperty.getMinimum());
+        String max = SwaggerUtils.toString(swaggerProperty.getMaximum());
+
+        if (swaggerProperty instanceof IntegerProperty
+                || swaggerProperty instanceof LongProperty) {
+            property.setMin(min == null ? null : min.replaceAll("\\.[0-9]+", ""));
```

Comment:
is a number with a dot valid? Can we do a Integer.parseInt then a toString instead to get the real int part?